### PR TITLE
Netdata fixes part 27

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
- Netdata fixes based on #22881 

---
## Summary by cubic
Hardened memory safety and correctness across core paths with bounds checks, safer arithmetic, and type fixes; added unit tests to cover edge cases. No behavior changes for valid inputs; prevents rare crashes, hangs, and overflows.

- **Bug Fixes**
  - Bounds and types: switched HTTP/WebSocket/ringbuffer indexes to size_t; updated logs and tests; `rbuf_find_bytes()` now returns offsets via size_t*.
  - Memory safety: clamped circular-buffer removals; guarded URL/DynCfg escape allocations against overflow; added overflow-safe math in `onewayalloc`; enforced dictionary key/value length bounds; validated interned string length; fixed two-way merge suffix bounds and initialized the “[x]” sentinel at startup.
  - Concurrency/lifecycle: made eval node IDs atomic; rejected invalid refcount releases from 0/DELETED; added CAS and underflow guards to object-state release/deactivation.
  - Parsers/algorithms: `holtwinters()` returns NAN on empty input; `duration_snprintf()` returns -3 on invalid unit; guarded KS2 index scaling and widened deltas to int64_t.
  - Data structures/utils: prevented double-linked-list self-append; early-return for zero-capacity splitters; `c_rhash` pointer getters now require pointer value type; bounded function-name scan with strnlen.
  - Tests: added dictionary length-bound tests and ringbuffer find-bytes tests.

<sup>Written for commit d27a3aa325440852cae1554bdd183cbcf0dc3df2. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/netdata/netdata/pull/22967?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/aclk/https_client.c [hunk 1]
```diff
@@ -241,7 +241,7 @@ const char *get_http_header_by_name(http_parse_ctx *ctx, const char *name)
 
 static int parse_http_hdr(rbuf_t buf, http_parse_ctx *parse_ctx)
 {
-    int idx, idx_end;
+    size_t idx, idx_end;
     char buf_key[HTTP_HDR_BUFFER_SIZE];
     char buf_val[HTTP_HDR_BUFFER_SIZE];
```
[the remark this answers is not in the document]
Comment (stelfrag) [reply src/aclk/https_client.c:246]: Valid, but this is a pre-existing bug: the identical separator logic is in master. This PR only changes idx/idx_end from int to size_t (native ringbuffer offset type) and does not alter the parsing; the cross-CRLF rbuf_pop happens the same way in both versions. Since it is orthogonal to the type fix and this PR's scope, we will address it in a separate PR: bound the separator search to idx_end (treat a separator at idx >= idx_end as missing), mirroring ws_client_parse_handshake_resp(). Tracked for follow-up.

**Review by thiagoftsm — APPROVED** [review]
PR is running as expected. LGTM!
